crowbar_framework/app/models/service_object.rb

```diff
@@ -1444,11 +1440,16 @@ def apply_role(role, inst, in_queue)
     proposal.save unless roles_to_remove.empty?
 
     update_proposal_status(inst, "success", "")
-    restore_to_ready(applying_nodes)
-    process_queue unless in_queue
     [200, {}]
+  rescue StandardError => e
+    @logger.fatal("apply_role: Uncaught exception #{e.message} #{e.backtrace.join("\n")}")
+    message = "Failed to apply the proposal: uncaught exception (#{e.message})"
+    update_proposal_status(inst, "failed", message)
+    [405, message]
```

Comment:
just wondering, here we return a message string (like in other places above), but just above the rescue block, on success we return an empty hash. I also couldn't find any consumer of that return message somehow